napari/utils/events/evented_model.py

```diff
@@ -391,14 +425,11 @@ def __eq__(self, other) -> bool:
         if not isinstance(other, EventedModel):
             return self.dict() == other
 
-        for f_name, eq in self.__eq_operators__.items():
-            if f_name not in other.__eq_operators__:
-                return False
-            if (
-                hasattr(self, f_name)
-                and hasattr(other, f_name)
-                and not eq(getattr(self, f_name), getattr(other, f_name))
-            ):
+        if set(self.__fields__) != set(other.__fields__):
+            return False
+        for f_name in self.__fields__:
+            eq = self.__eq_operators__[f_name]
+            if not eq(getattr(self, f_name), getattr(other, f_name)):
```

Comment:
@brisvag, I have fixed it differently. There is no need to compare properties.

Reply:
Assuming the equal fields mean equal properties I guess?

Reply:
In the previous implementation, we check only a subset of another object. 
Maybe we need to add a check for class equality?

Reply:
If class is the same and fields are the same, that sounds like they are the same, yes!

Reply:
You could have another class that implements the same interface, but I'm unsure if we would like to support it.